### PR TITLE
Plugin behaviour with System implementation

### PR DESCRIPTION
**Pull request by brentonannan**
The `Plugin` behaviour requires implementation of functions for fetching (as well as potentially transforming and ... other things) configuration values from a specific remote.

### CODE REVIEW

File: lib/configurator/system.ex [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  config :configurator, :system_example,
+    source: {Configurator.System, "SYSTEM_ENV_VAR_NAME"},
+    target: {:application, :key}
+  ```
+  """
+  use Configurator.Plugin
+
+  @argument_error "`Configuration.System` requires a keyword list with an `:env` key"
+  def fetch(args) when is_list(args) do
```
Comment (wiserfirst) [inline lib/configurator/system.ex]: I don't quite understand why you want the `args` to be a keyword list. What benefit does that bring over just passing Env var as a string?

File: test/support/plugin.ex [hunk 2]
```diff
@@ -0,0 +1,7 @@
+defmodule Configurator.Test.Plugin do
+  use Configurator.Plugin
+
+  def fetch(args) do
```
Comment (wiserfirst) [inline test/support/plugin.ex]: where is this `fetch` function used?

File: lib/configurator/system.ex (hunk 1 above)
Comment (brentonannan) [inline lib/configurator/system.ex]: Passing `args` as a keyword list allows any number of args to be passed to any implementation of `Configurator.Plugin`; having specific arguments would require a less generic type signature for the callback.

File: test/support/plugin.ex (hunk 2 above)
Comment (brentonannan) [inline test/support/plugin.ex]: It's a mock implementation of `Plugin` for testing. It'll be used where fetch is used in resolving named configs, but its use is out of the scope of this PR.

**Pull request merged**